settings/security/third-party/saml.rst

```diff
@@ -19,81 +22,78 @@ Configure SAML in Zammad
-
-**IDP certificate fingerprint:**
-Optionally use the fingerprint instead of the certificate itself for cert validation.
-Because sha1 is used for the fingerprint, this option is not preferred.
-Do not set this option at all if using the "certificate" field above.
+IDP SSO target URL:
+   ``https://{keycloak-FQDN}/auth/realms/master/protocol/saml``
+   
+   .. note:: This will vary according to your IdP documentation
+
+IDP certificate: (base64 encoded Public Cert from the SAML IdP)
+   .. code-block:: html
+   
+      MIIDazCCAlOgAwIBAgIUFIWdqF4bXK5ajmxEyNJ03uK+5UQwDQYJKoZIhvcNAQELBQAwRTELMAkGA1UEBhMCVVMxEzARBgNVBAgMClNvbWUtU3RhdGUxITAfBgNVBAoMGEludGVybmV0IFdpZGdpdHMgUHR5IEx0ZDAeFw0xOTA5MDQxNjMxNDlaFw0xOTEwMDQxNjMxNDlaMEUxCzAJBgNVBAYTAlVTMRMwEQYDVQQIDApTb21lLVN0YXRlMSEwHwYDVQQKDBhJbnRlcm5ldCBXaWRnaXRzIFB0eSBMdGQwggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQCYbCP4kxftE00X/jtZ057eJLfWbkKaNonbUNyvURL9+1CY8Vg4y5941qZQ+Wib6Em+2TzCVCg4v/6oOAgFtnQApd+h8J+PgQ1iDj369oOhI7I0AwU37fNkGWrQWZi6GqV5xWlApb+3dwY2ag9dKF41n+lf0eFzWXGIaYPRMq3tVt9oz0Jxosuz/aYX2ktEydQTSBng+smUq5vdlnRTqKKu3MFCeDVqb6f9FtXz7xKV/bwcMepz0eOw8TKjfWK3Y4OFKjfHhHFG+ii8eNOFmmHn767K96819v86ehUDpY/h+lKnrnxjv/115Uu0zrB2OAfjvcQZNmfnA/rKL7UjsGl3AgMBAAGjUzBRMB0GA1UdDgQWBBRnqILKxsNGS4iX79uTcjXBBm75XTAfBgNVHSMEGDAWgBRnqILKxsNGS4iX79uTcjXBBm75XTAPBgNVHRMBAf8EBTADAQH/MA0GCSqGSIb3DQEBCwUAA4IBAQBRST4IB5O67j9ziSIzaJhhzKzu7QWWeSVcLECyfJ5iik0BMvcC3YB4rSoHo4nWgCCb+EGIaqpotXV5dK2zfCHe85bQCrc5xFZmiKCmN2iLvkF3xc9twlw8yvxSBLO6rqceVNlwJwVVsW/63v3+GXEXm0y9yPYjyr6e/VE6AxAv650dccMThxL/5ZQyceE9qSNMPk2C6kiBTv/ZKosCratsiGOWok56WyCzJyag4I93IEiFRaZlWvtJMDBXYAbgPalwm9MPoU70n6K8Bf8L+Tekt6v1ny8Iv14Whb6l5XMr/r4cqwv2DNwq1xMh59WITgdJwUNB5uLLldm40RRBJtb0
+   
+   .. note::
+      - The above certificate is just an example, you should use a proper certificate from your IdP!
+      - Note that this is generally **not** the same cert used by TLS in the HTTPS connection to the IdP.
+
+   .. hint:: **ADFS users**
+      
+      Above certificate refers to the to the signing certificate.
+
+   In Keycloak, this certificate can be found by going to:
+      
+      - Keycloak Admin: Realm settings -> keys -> RSA -> "certifcate" popup
+      - Paste as a single line string with no line breaks.
+      - Do not set this option at all if using the "certificate fingerprint" for cert validation.
+
+IDP certificate fingerprint:
+   Optionally use the fingerprint instead of the certificate itself for cert validation.
+   Because sha1 is used for the fingerprint, this option is not preferred.
+   Do not set this option at all if using the "certificate" field above.
 
 **Name Identifier Format:**
-
-``urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress``
-
-
-Configured example:
-
-.. image:: /images/settings/security/third-party/saml/zammad_connect_saml_thirdparty.png
-   :alt: Zammad SAML Example
-
+   ``urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress``
```

Comment:
I've removed a lot of this, because large parts of it felt:

* redundant (repeating info already found in the screenshot)
* unnecessary (very long sample signing certificate)
* not authoritative; _i.e.,_ particular to some people's setups but not others, without explaining why.

Here is my rewrite of the entire section:

![Screenshot_20200728_235108](https://user-images.githubusercontent.com/12194123/88689842-91519d80-d12d-11ea-835e-c4586c852d2b.png)

Does that look correct/good to you?

#### Notes

I think the original docs specified the wrong place to copy the Keycloak signing certificate from? Happy to discuss over a screenshare.

Reply:
Looking perfectly fine to me, thank you!


> I think the original docs specified the wrong place to copy the Keycloak signing certificate from? Happy to discuss over 
> a screenshare.

I'm sorry, I can't answer this question.
The documentation has been provided by a user that was helping us with SAML.

Reply:
Okay. Maybe we can ask Thorsten at tomorrow's standup? Looks like he wrote the `lib/omniauth/saml_database.rb` file.

Reply:
So I spoke to Thorsten, and in the end, we decided to revert it back to the original since customers have been using the existing documentation without any (reported) issues.